docs/test_skip_expiry_design.md

```diff
@@ -0,0 +1,465 @@
+
+This document proposes a solution to prevent test skips from being forgotten indefinitely by introducing a category-based skip management system. Skips are classified into two categories: **permanent** (indefinite skips without expiry dates) and **temporary** (time-bound skips with mandatory expiry dates). This ensures that temporary skips are revisited and either fixed or explicitly renewed, while permanent skips are properly justified.
+
+---
+
+## Table of Contents
+
+1. [Background](#background)
+2. [Problem Statement](#problem-statement)
+3. [Proposed Approaches](#proposed-approaches)
+   - [Approach 1: Inline Expiry Date Field](#approach-1-inline-expiry-date-field)
+   - [Approach 2: Expiry Management via Separate Configuration](#approach-2-expiry-management-via-separate-configuration)
+4. [Comparison and Recommendation](#comparison-and-recommendation)
+5. [Detailed Design (Recommended Approach)](#detailed-design-recommended-approach)
+6. [Implementation Considerations](#implementation-considerations)
+7. [Migration Strategy](#migration-strategy)
+8. [Testing Strategy](#testing-strategy)
+9. [Future Enhancements](#future-enhancements)
+
+---
+
+## Background
+
+The `tests_mark_conditions.yaml` file serves as a centralized location for defining test skips and their conditions. The pytest plugin `conditional_mark` processes this file to automatically skip tests based on platform, topology, release version, and other conditions.
+
+While this consolidation improves maintainability, tests can be skipped and forgotten, leading to:
+- Tests remaining skipped long after issues are resolved
+- Technical debt accumulation
+- Reduced test coverage
+- Loss of visibility into why tests were skipped
+
+---
+
+## Problem Statement
+
+**Goal**: Implement a category-based skip management system to ensure that skipped tests are properly classified and periodically reviewed:
+1. **Permanent skips**: For fundamental limitations (e.g., ASIC not supported) without expiry dates
+2. **Temporary skips**: For transient issues (e.g., bugs being fixed) with mandatory expiry dates
+3. Tests must be re-enabled if the underlying issue is resolved
+4. Temporary skips must be updated with a new expiry date if the issue persists
+
+**Requirements**:
+- Two distinct skip categories: permanent and temporary
+- Enforce expiry dates for temporary skips
+- Validate category-specific allowed reasons
+- Clear handling of expired skips (force test to run or fail with explicit message)
+- Timezone-aware date handling for consistency across global teams
+- Minimal disruption to existing workflows
+- Clear reporting of expired and miscategorized skips
+
+---
+
+## Proposed Approaches
+
+### Approach 1: Inline Category and Expiry Date Fields
+
+Add `category` and `expiry_date` fields directly within each skip/xfail definition in the YAML file, with skip categories defined at the top level.
+
+#### Structure Example
+
+```yaml
+# Define skip categories at the top of the file
```

Comment:
Should we maintain the categories in a separate file and restrict editing to specific roles? This would allow tighter control over category changes.